internal/server/stat_point.go

```diff
@@ -200,47 +200,113 @@ func (s *Server) GetStatCollection(
 	// Initialize with nil to help check if data is in mem-cache. The nil field
 	// will be populated with empty pb.ObsCollection struct in the end.
 	for _, sv := range statVars {
-		result.Data[sv] = nil
+		result.Data[sv] = &pb.SourceSeries{}
 	}
 
-	var rowList bigtable.RowList
-	var keyTokens map[string]string
-	if svobsMode {
-		rowList, keyTokens = buildStatCollectionKeySvObs(parentPlace, childType, date, statVars)
-	} else {
-		// Read triples for statistical variable.
-		triplesRowList := buildTriplesKey(statVars)
-		triples, err := readTriples(ctx, s.store, triplesRowList)
+	// Mapping from stat var to a list of dates. Need to fetch cache data for any
+	// <sv, date> pair.
+	sv2dates := map[string][]string{}
+
+	// If date is not given, get the latest dates and use them to fetch
+	// stats.
+	if date == "" {
+		rowList, keyTokens := buildStatCollectionKeySvObs(parentPlace, childType, "", statVars)
+		dateCache, err := readStatCollection(ctx, s.store, rowList, keyTokens)
 		if err != nil {
 			return nil, err
 		}
-		statVarObject := map[string]*StatisticalVariable{}
-		for statVar, triplesCache := range triples {
-			if triplesCache != nil {
-				statVarObject[statVar], err = triplesToStatsVar(statVar, triplesCache)
-				if err != nil {
-					return nil, err
+		// Two dates are computed here. One is the latest date from all sources and
+		// the other is the date corresponding to the most place entries.
+		for sv, data := range dateCache {
+			latestDate := dateCount{}
+			mostCount := dateCount{}
+			if data != nil {
+				for _, cohort := range data.SourceCohorts {
+					for date, c := range cohort.Val {
+						count := int(c)
+						if count > mostCount.count || (count == mostCount.count && date > mostCount.date) {
+							mostCount = dateCount{date, count}
+						}
+						if date > latestDate.date || (date == latestDate.date && count >= latestDate.count) {
+							latestDate = dateCount{date, count}
+						}
+					}
 				}
 			}
+			// Lastest date is always needed.
+			sv2dates[sv] = []string{latestDate.date}
+			// The date with most place count is needed too.
+			if mostCount.count > latestDate.count {
+				sv2dates[sv] = append(sv2dates[sv], mostCount.date)
+			}
```

Comment:
Should this block also be inside the if data != nil check?

Reply:
good catch, "continue" earlier with data == nil